Improve cloud backport script

Algunenano opened the pull request:
<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Improve cloud backport script

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->


> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>Modify your CI run</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ] <!---ci_include_fast--> Fast test
- [ ] <!---ci_include_integration--> Integration Tests
- [ ] <!---ci_include_stateless--> Stateless tests
- [ ] <!---ci_include_stateful--> Stateful tests
- [ ] <!---ci_include_unit--> Unit tests
- [ ] <!---ci_include_performance--> Performance tests
- [ ] <!---ci_include_asan--> All with ASAN
- [ ] <!---ci_include_tsan--> All with TSAN
- [ ] <!---ci_include_analyzer--> All with Analyzer
- [ ] <!---ci_include_azure --> All with Azure
- [ ] <!---ci_include_KEYWORD--> Add your option here

#### Exclude tests:
- [ ] <!---ci_exclude_fast--> Fast test
- [ ] <!---ci_exclude_integration--> Integration Tests
- [ ] <!---ci_exclude_stateless--> Stateless tests
- [ ] <!---ci_exclude_stateful--> Stateful tests
- [ ] <!---ci_exclude_performance--> Performance tests
- [ ] <!---ci_exclude_asan--> All with ASAN
- [ ] <!---ci_exclude_tsan--> All with TSAN
- [ ] <!---ci_exclude_msan--> All with MSAN
- [ ] <!---ci_exclude_ubsan--> All with UBSAN
- [ ] <!---ci_exclude_coverage--> All with Coverage
- [ ] <!---ci_exclude_aarch64--> All with Aarch64
- [ ] <!---ci_exclude_KEYWORD--> Add your option here

#### Extra options:
- [ ] <!---do_not_test--> do not test (only style check)
- [ ] <!---no_merge_commit--> disable merge-commit (no merge from master before tests)
- [ ] <!---no_ci_cache--> disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ] <!---batch_0--> 1
- [ ] <!---batch_1--> 2
- [ ] <!---batch_2--> 3
- [ ] <!---batch_3--> 4

<details>

Algunenano committed:
Always try to backport to the cloud (v24.3-must-backport)

Algunenano committed:
Create cloud backports for all pr-must-backport PRs

Even if pr-must-backport-cloud is not declared

robot-ch-test-poll commented:
<!-- automatic status comment for PR #63282 from Algunenano/ClickHouse:backport_cloud -->
*This is an automated comment for commit d5c66a9aac6ba8ae951fcfc0125aac0a8bf3c2f4 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63282/d5c66a9aac6ba8ae951fcfc0125aac0a8bf3c2f4/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

Felixoid commented on tests/ci/cherry_pick.py:
```diff
@@ -402,11 +402,21 @@ def receive_release_prs(self):
+            ]
+
+            # Forcefully add also non cloud backports
+            self.labels_to_backport += [Labels.MUST_BACKPORT]
```
This is the only thing I'd like to fix. Let's make `--must-create-backport-label` a list and pass it to `Backport.must_create_backport_labels` instead of `Backport.must_create_backport_label`.

Maybe, I'd think of the `--aggressive-backport`, but your solution is very elegant.

Felixoid replied:
Or simpler. Since we always use `Labels.MUST_BACKPORT`, we can add `--extra-create-backport-label`. Empty by default, adding `MUST_BACKPORT_CLOUD`

Algunenano replied:
Changed to a list. The simpler option would be to remove options and have just 2 modes: public and private. No need to support custom tags, repo names, etc

Algunenano committed:
Change must-create-backport-label to support a list

python3 cherry_pick.py --repo ClickHouse/clickhouse-private  --from-repo ClickHouse/ClickHouse --must-create-backport-label pr-must-backport-cloud pr-must-backport --backport-created-label pr-backports-created-cloud

Algunenano committed:
Fix dev

Felixoid commented on tests/ci/cherry_pick.py:
```diff
@@ -473,12 +480,18 @@ def process_backports(self):
-        else:
+
+        for label in self.must_create_backport_labels:
+            # We backport any vXXX-must-backport to all branches of the fetch repo (better than no backport)
+            if label in pr_labels or self._fetch_from:
+                branches = [
+                    ReleaseBranch(br, pr, self.repo, self.backport_created_label)
+                    for br in self.release_branches
+                ]  # type: List[ReleaseBranch]
+                found = True
```
The `found` is not used

```suggestion
            # We backport any vXXX-must-backport to all branches of the _fetch_from repo (better than no backport)
            if label in pr_labels or self._fetch_from:
                branches = [
                    ReleaseBranch(br, pr, self.repo, self.backport_created_label)
                    for br in self.release_branches
                ]  # type: List[ReleaseBranch]
```

Algunenano committed:
Update cherry_pick.py

Algunenano committed:
Merge remote-tracking branch 'blessed/master' into backport_cloud

Felixoid commented:
Let's test how it will work! I'll launch the backport CI from cloud
